api/app/src/routes/util.ts

```diff
@@ -13,13 +15,45 @@ export const asyncHandler =
   ) =>
   async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
+      analyzeRoute(req);
       await f(req, res, next);
     } catch (err) {
       console.error(err);
       next(err);
     }
   };
 
+export const analyzeRoute = (req: Request): void => {
+  let cxId;
+
+  const reqCxId = getCxId(req);
+  if (reqCxId) cxId = reqCxId;
+
+  const headerCxId = getCxIdFromHeaders(req);
+  if (headerCxId) cxId = headerCxId;
+
+  if (cxId) {
+    const isMedical = req.baseUrl.includes("medical");
+
+    let reqUrl = req.baseUrl;
+    const hasPath = req.route.path.split("/")[1];
+
+    if (hasPath) {
+      reqUrl = reqUrl.concat(req.route.path);
+    }
+
+    analytics({
+      distinctId: cxId,
+      event: EventTypes.query,
+      properties: {
+        method: req.method,
+        url: reqUrl,
+        apiType: isMedical ? ApiTypes.medical : ApiTypes.devices,
```

Comment:
There are endpoints that are not MAPI or DAPI, like `/internal/` and MAPI related ones that are not "medical", like `/oauth` - not sure how we want to approach these.

Reply:
Good point updated with a new solution